Composer/packages/form-dialogs/src/utils/constants.ts

```diff
@@ -2,3 +2,5 @@
 // Licensed under the MIT License.
 
 export const jsPropertyListClassName = 'jsPropertyListClassName';
+
+export const nameRegex = /^[a-zA-Z0-9-_]+$/;
```

Comment:
I see this in the SDK schema for AgeEntityRecognizer.  I wonder if the sdk schema should define a pattern for a valid property name rather than string?  Or maybe this is too meta for a JSON schema.

"$kind": {
          "title": "Kind of dialog object",
          "description": "Defines the valid properties for the component you are configuring (from a dialog .schema file)",
          "type": "string",
          "pattern": "^[a-zA-Z][a-zA-Z0-9.]*$",
          "const": "Microsoft.ActivityTemplate"
        },